bitcoin/src/blockdata/script/witness_program.rs

```diff
@@ -37,33 +44,42 @@ impl WitnessProgram {
         P: TryInto<PushBytesBuf>,
         <P as TryInto<PushBytesBuf>>::Error: PushBytesErrorReport,
     {
+        let program =
+            program.try_into().map_err(|error| Error::InvalidLength(error.input_len()))?;
+        WitnessProgram::from_bytes(version, program.as_bytes())
+    }
+
+    /// Creates a new witness program, copying the content from the given byte slice.
+    pub fn from_bytes(version: WitnessVersion, bytes: &[u8]) -> Result<Self, Error> {
```

Comment:
I really don't like this, let's just change the bound to `P: AsRef<[u8]>`. All types that impl `TryInto<PushBytesBuf>` should already impl `AsRef<[u8]>` already (if not it should be added unless I'm missing something).

Reply:
Fixed! `WitnessProgram::new` has deleted, and `WitnessProgram::from_bytes` has been renamed to take its place.